### DISCUSSION TITLE
Strange error when mirroring Orbit repo

### DISCUSSION THREAD

**Discussion by andrew-tram** [eclipse-orbit/orbit-simrel#34] [Q&A] [answered]
I just have a simple tycho mirroring job that captures the content of Orbit Aggregation and creates an update-site from it locally.

Using the following repo:

```
		<repository>
			<layout>p2</layout>
			<id>repo</id>
			<url>https://download.eclipse.org/tools/orbit/simrel/orbit-aggregation/release/4.31.0</url>
		</repository>
```

I get the following error:

```
[ERROR]   Missing requirement: org.eclipse.orbit.maven.org.apache.httpcomponents.client5.httpclient5-win.feature.group 4.31.0.v20231203-1619 requires 'org.eclipse.equinox.p2.iu; org.apache.httpcomponents.client5.httpclient5-win [5.2.3.v20231203-1619,5.2.3.v20231203-1619]' but it could not be found
```

Which is strange because the plugin is there, the metadata is present in the content.jar, and the jar itself is physically in the repository location locally (and remotely via Jenkins, same error). 

![image](https://github.com/eclipse-orbit/orbit-simrel/assets/106191140/91c1cf86-2b63-4e52-b800-742b962af1df)

I'm not sure why this is happening as I did this same tasks with 4.30 without issues.

Any pointers would be greatly appreciated.

**Comment by merks** [comment]
This bundle has a filter that it's only applicable for win32:

![image](https://github.com/eclipse-orbit/orbit-simrel/assets/208716/b936df39-5bad-4b49-a996-d8e2980fdca6)

But that's always been the case and was also the case for the 4.30 release.

Perhaps you are using a new version of Tycho and it's more properly checking/respecting filters?

Note that the filter also looks for `org.eclipse.orbit.maven.bnd.buildtime=true` and I use the that in some places to avoid resolution failures:

![image](https://github.com/eclipse-orbit/orbit-simrel/assets/208716/fcefc5f8-e06b-4e0b-89cb-e45a4ee53e10)

Keep in mind that every simple repository comes also in zipped form:

- https://www.eclipse.org/downloads/download.php?file=/tools/orbit/simrel/orbit-aggregation/release/4.31.0/Orbit-Aggregation-Updates-4.31.0.zip
- https://download.eclipse.org/tools/orbit/simrel/orbit-aggregation/release/4.31.0/Orbit-Aggregation-Updates-4.31.0.zip

**Comment by andrew-tram** [comment, marked as answer]
@merks , 

Adding `<org.eclipse.orbit.maven.bnd.buildtime>true</org.eclipse.orbit.maven.bnd.buildtime>` seemed to do the trick.

I was building locally on Windows to make sure it wasnt just the Linux build machine issue to check for platform filtering. And I went from Tycho 4.0.5 to 4.0.6, swapped back and that didnt make a difference either.

Is `org.eclipse.orbit.maven.bnd.buildtime` a new thing added in 4.31?

**Reply by merks** [reply]
It's been there since the start because I had problems aggregating/building as well.  This is the only bundle with such a filter...
